exchanges/binanceus/binanceus.go

```diff
@@ -0,0 +1,1805 @@
+	endTimeT := time.UnixMilli(int64(endTime))
+
+	hundredDayBefore := time.Now()
+	hundredDayBefore.Sub(time.UnixMilli(int64((time.Hour * 24 * 10) / time.Millisecond)))
+	if !(startTimeT.Before(hundredDayBefore)) || !startTimeT.After(time.Now()) {
+		params.Set("startTime", strconv.Itoa(int(startTime)))
+	}
+	if !(endTimeT.Before(hundredDayBefore)) || !endTimeT.After(time.Now()) {
+		params.Set("startTime", strconv.Itoa(int(endTime)))
+	}
+	return resp.Transfers, b.SendAuthHTTPRequest(ctx,
+		exchange.RestSpotSupplementary,
+		http.MethodGet,
+		subaccountTransferHistory,
+		params,
+		spotDefaultRate,
+		resp)
+}
+
+// ExecuteSubAccountTransfer to execute sub-account asset transfers.
+// POST /wapi/v3/sub-account/transfer.html (HMAC SHA256)
+// Use this endpoint to execute sub-account asset transfers.
+func (b *Binanceus) ExecuteSubAccountTransfer(ctx context.Context, arg *SubaccountTransferRequestParams) (*SubaccountTransferResponse, error) {
+	params := url.Values{}
+	var response SubaccountTransferResponse
+	if !MatchesEmailPattern(arg.FromEmail) {
+		return nil, errUnacceptableSenderEmail
+	}
+	if !MatchesEmailPattern(arg.ToEmail) {
+		return nil, errUnacceptableReceiverEmail
+	}
+	if len(arg.Asset) <= 2 {
+		return nil, errInvalidAssetValue
+	}
+	if arg.Amount <= 0.0 {
+		return nil, errInvalidAssetAmount
+	}
+	params.Set("fromEmail", arg.FromEmail)
+	params.Set("toEmail", arg.ToEmail)
+	params.Set("asset", arg.Asset)
+	params.Set("amount", fmt.Sprintf("%f", arg.Amount))
+	params.Set("recvWindow", "5000")
+	params.Set("timestamp", fmt.Sprintf("%d", time.Now().UnixMilli()))
+	return &response, b.SendAuthHTTPRequest(ctx, exchange.RestSpotSupplementary, http.MethodPost, subaccountTransfer, params, spotDefaultRate, &response)
+}
+
+// GetSubaccountAssets to fetch sub-account assets.
+// GET /wapi/v3/sub-account/assets.html (HMAC SHA256)
+// Use this endpoint to fetch sub-account assets.
+func (b *Binanceus) GetSubaccountAssets(ctx context.Context, email string) (*SubAccountAssets, error) {
+	var resp SubAccountAssets
+	if !MatchesEmailPattern(email) {
+		return nil, errNotValidEmailAddress
+	}
+	params := url.Values{}
+	timestamp := time.Now().UnixMilli()
+	params.Set("timestamp", fmt.Sprintf("%d", timestamp))
+	params.Set("email", email)
+	//
+	return &resp, b.SendAuthHTTPRequest(ctx,
+		exchange.RestSpotSupplementary, http.MethodGet,
+		subaccountAssets, params,
+		spotDefaultRate,
+		&resp)
+}
+
+// Trade Order Endpoints
+
+// GetOrderRateLimits Get the current trade order count rate limits for all time intervals.
+// GET /api/v3/rateLimit/order (HMAC SHA256)
+// Get the current trade order count rate limits for all time intervals.
+// INPUTS: recvWindow <= 60000
+func (b *Binanceus) GetOrderRateLimits(ctx context.Context, recvWindow uint) ([]OrderRateLimit, error) {
+	params := url.Values{}
+	timestamp := time.Now().UnixMilli()
+	params.Set("timestamp", strconv.Itoa(int(timestamp)))
+	params.Set("recvWindow", strconv.Itoa(30000))
```

Comment:
recvWindow param is not being handled

Reply:
I don't get this message.
I thought for some endpoints the maximum endpoint recvWindow is 60,000; that's why I use 30,000 if that is the case!?